spec/javascript/testUtils/wrapperUtil.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import Context from '~/components/supplemental_application/context'
+import { Form } from 'react-final-form'
+
+const formNode = (application, formToChildrenFunc) => (
+  <Form
+    onSubmit={() => null}
+    initialValues={application}
+    render={({ form }) => (
+      <form noValidate>
+        {formToChildrenFunc(form)}
+      </form>
+    )}
+  />
+)
+
+const diveThroughFormWrappers = (wrapper) =>
+  wrapper
+    .find('form')
+    .children()
+    .dive()
```

Comment:
this is really great to be able to "dive" through a couple of layers to be able to keep using shallow render 🎉

Reply:
It is!

It can get confusing if used too much though. I think putting it in named functions like this or adding comments that explain what you're diving into is important. At my last job people joked that if your test didn't work just keep adding dive() until it does. That's bad lol